mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_jwt.go

```diff
@@ -0,0 +1,145 @@
+
+func GoogleEphemeralServiceAccountJwt() ephemeral.EphemeralResource {
+	return &googleEphemeralServiceAccountJwt{}
+}
+
+type googleEphemeralServiceAccountJwt struct {
+	providerConfig *fwtransport.FrameworkProviderConfig
+}
+
+func (p *googleEphemeralServiceAccountJwt) Metadata(ctx context.Context, req ephemeral.MetadataRequest, resp *ephemeral.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_account_jwt"
+}
+
+type ephemeralServiceAccountJwtModel struct {
+	Payload              types.String `tfsdk:"payload"`
+	ExpiresIn            types.Int64  `tfsdk:"expires_in"`
+	TargetServiceAccount types.String `tfsdk:"target_service_account"`
+	Delegates            types.Set    `tfsdk:"delegates"`
+	Jwt                  types.String `tfsdk:"jwt"`
+}
+
+func (p *googleEphemeralServiceAccountJwt) Schema(ctx context.Context, req ephemeral.SchemaRequest, resp *ephemeral.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Produces an arbitrary self-signed JWT for service accounts.",
+		Attributes: map[string]schema.Attribute{
+			"payload": schema.StringAttribute{
+				Required:    true,
+				Description: `A JSON-encoded JWT claims set that will be included in the signed JWT.`,
+			},
+			"expires_in": schema.Int64Attribute{
+				Optional:    true,
+				Description: "Number of seconds until the JWT expires. If set and non-zero an `exp` claim will be added to the payload derived from the current timestamp plus expires_in seconds.",
+			},
+			"target_service_account": schema.StringAttribute{
+				Description: "The email of the service account that will sign the JWT.",
+				Required:    true,
+			},
+			"delegates": schema.SetAttribute{
+				Description: "Delegate chain of approvals needed to perform full impersonation. Specify the fully qualified service account name.",
+				Optional:    true,
+				ElementType: types.StringType,
+			},
+			"jwt": schema.StringAttribute{
+				Description: "The signed JWT containing the JWT Claims Set from the `payload`.",
+				Computed:    true,
+				Sensitive:   true,
+			},
+		},
+	}
+}
+
+func (p *googleEphemeralServiceAccountJwt) Configure(ctx context.Context, req ephemeral.ConfigureRequest, resp *ephemeral.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	pd, ok := req.ProviderData.(*fwtransport.FrameworkProviderConfig)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *fwtransport.FrameworkProviderConfig, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	p.providerConfig = pd
+}
+
+func (p *googleEphemeralServiceAccountJwt) Open(ctx context.Context, req ephemeral.OpenRequest, resp *ephemeral.OpenResponse) {
+	var data ephemeralServiceAccountJwtModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	payload := data.Payload.ValueString()
+
+	if !data.ExpiresIn.IsNull() {
+		expiresIn := data.ExpiresIn.ValueInt64()
+		if expiresIn != 0 {
```

Comment:
Should a user be able to set an expires_in of 0?

I suspect this line of code is something adapted from the original SDK code, where we need to avoid zero values. However in the plugin-framework there are no zero values, only Nulls if a field is unset, so this code is handling the case of a user setting a value of 0 in their config.

I think we should:
- Add a validator to the expires_in field that asserts it must be >0
- double check if the if logic here is necessary after that

Reply:
new validator added, also ran locally and no need for the if logic since validator is triggered before ever reaching the open method:
```hcl
│ Error: Invalid Integer Value
│ 
│   with ephemeral.google_service_account_jwt.test_jwt,
│   on main.tf line 1, in ephemeral "google_service_account_jwt" "test_jwt":
│    1: ephemeral "google_service_account_jwt" "test_jwt" {
│ 
│ Value must be greater than 0, got: 0
```

Reply:
Thanks!